src/lib/session-authenticator.js

```diff
@@ -0,0 +1,72 @@
+/* global Auth0Lock */
+/**
+ * Encapsulates Authentication Functionality For a Client Side Application
+ */
+export default class SessionAuthenticator {
+  /**
+   * @constructor
+   */
+  constructor (env) {
+      this.sessionUrl = serverEnv.SESSION_URL
+      this.tokenUrl = serverEnv.TOKEN_URL
+      this.endpoints = serverEnv.ENDPOINTS
+  }
+
+  /**
+   * Whether or not this authentication module supports login
+   * @return {Boolean} false for server side auth
+   */
+  enableLogin() {
+    return false
+  }
+
+
+  session () {
+    return new Promise((resolve,reject) => {
+      window.fetch(this.sessionUrl).then((resp) => {
+        if (! resp.ok) {
+          reject(resp.code)
+        } else {
+          resolve(resp.json())
+        }
+      })
+    })
```

Comment:
I'm wondering if and how the `session` promise gets resolved if `fetch` itself is rejected. Do we need to have a `catch` block for it?

Reply:
good point, will fix